R/derive_interp_records.R

```diff
@@ -0,0 +1,74 @@
+#' Derive interpolated rows for the CDC charts (>=2 yrs old)
+#'
+#' Derive interpolated rows for the CDC charts (>=2 yrs old) by age in days
+#' for the following parameters: HEIGHT, WEIGHT and BMI
+#'
+#' @param dataset Input metadataset
+#'
+#'   The variables `AGE`, `AGEU`, `SEX`, `L`, `M`, `S` are expected to be in the dataset
+#'
+#'   For BMI the additional variables `P95` and `Sigma` are expected to be in the dataset
+#'
+#'   Note that `AGE` must be in days so that `AGEU` is equal to `"DAYS"`
+#'
+#' @param parameter CDC/WHO metadata parameter
+#'
+#' *Permitted Values*: `"WEIGHT"`, `"HEIGHT"` or `"BMI"` only - Must not be `NULL`
+#'
+#'   e.g. `parameter = "WEIGHT"`,
+#'        `parameter = "HEIGHT"`,
+#'   or   `parameter = "BMI"`.
+#'
+#' @return The input dataset plus additional interpolated records
+#'
+#' @family metadata
+#'
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' cdc_htage <- admiralpeds::cdc_htage %>%
+#'   mutate(
+#'     SEX = case_when(
+#'       SEX == 1 ~ "M",
+#'       SEX == 2 ~ "F",
+#'       TRUE ~ NA_character_
+#'     ),
+#'     AGE = round(AGE * 30.4375)
+#'   ) %>%
+#'   # Interpolate the AGE by SEX
+#'   group_by(SEX) %>%
+#'   # Ensure first that Age unit is "DAYS"
+#'   do(derive_interp_records(., parameter = "HEIGHT")) %>%
+#'   ungroup() %>%
+#'   # Keep patients >= 2 yrs only - Remove duplicates for 730 Days old which
+#'   # must come from WHO metadata only
+#'   filter(AGE >= 730.5)
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset, parameter = "WEIGHT") {
```

Comment:
```suggestion
derive_interp_records <- function(dataset,
                                  parameter) {
```
parameter shouldn't have a default value and then more consistent with the other functions. also if you add `by_vars` as per other comment better to show these arguments each on new line as looks better in the reference page

Reply:
Done

Reply:
@Fanny-Gautier you still left the default values:

```
derive_interp_records <- function(dataset,
                                  by_vars = exprs(SEX),
                                  parameter = "WEIGHT") {
```

it should be:

```
derive_interp_records <- function(dataset,
                                  by_vars = NULL,
                                  parameter) {
```

Reply:
Implemented, I misunderstood the first comment.